### DISCUSSION TITLE
Can it be automatically grouped by Tags, e.g. by date[s] (initially bookmarked, last used in session)?

### DISCUSSION THREAD

**Discussion by staalman** [GarinZ/link-map#16] [Ideas]
Can it be automatically grouped according to a **<tag>** either manually entered, or e.g. system tags like e.g. **dates** (e.g. initially bookmarked date , date of last used in a browser session)? 

I saw the "grouped by site" mentioned as an issue, although it's not an issue but a (good) idea.
In a similar fashion, I would like to see sorting by initial-list-date and last-used-date as this helps to find recently used ones back in the taxonomy that you created based on a different structure than <time>.

In more general terms, adding tags as a way to provide multi-dimensional groupings would be very, very useful as the single taxonomy is - no matter how great - sometimes falling short.

PS Great that you started this initiative, because I was also experiencing the drawbacks caused by a lack of support

**Comment by GarinZ** [comment]
Hi @staalman,
Your suggestion has inspired me a lot. In fact, I am currently designing this automatic grouping function. According to my current design, the grouping rules will be configurable, not just based on domain names. Additionally, I will add the option to group by date based on your suggestion.

Furthermore, you mentioned the sorting function, which I also find very useful. Since the implementation of automatic grouping and sorting in a tree structure may be complex, I may release them separately in different versions.

**Reply by staalman** [reply]
Sounds great! I'm *really* happy to see this initiative as Tabs Outliner is an excellent tool, but has it's shortcomings, most importantly the lack of support. Changing/evolving security related chrome policies are often in the way of it. Now that I see you actually like feedback, input and ideas, I will look for my notes on TO i collected over the years (I think I'm using it from around a decade now).
On the feedback page of TO you can also browse and pick up some good ideas. I will post another one in a new thread in a minute.

Regarding "grouping by tag": if the tags would be in a list from which you can select, the tree could be filtered 'live' by only showing the taxonomy with entries that have that tag. It will allow the multi-dimensional taxonomy concept to be used because of the tags, while still sticking to one main taxonomy concept, which is by default probably 'last used date" like with TO, unless you start naming the branches and sorting it.
One other part of the grouping in TO is the difference made between "Groups" and "Windows", where "groups" do allow clustering of any sub branch without opening any windows/tabs when double-clicking it, while "Windows" does allow the same, but then automatically opens by default all the entries as tabs that it is grouping together. That's a practical feature, also when using tags.

**Reply by staalman** [reply]
Just to give you the idea: a couple of years ago, I posted the following as Tabs Outliner review:
------------------------------------------------------------
1. Tab Group support in Chrome would be excellent.
2. Multi Device - Single Account usage.
Improvement in this area requires some thought: no one is using just one PC these days; and e.g. Chrome - is when logged in - already syncing bookmarks between devices; that's  a step TO's has to make too. Especially relevant now TO is my history-session manager and Tabs manager in one.
This means: 
  a. sharing Tabs across devices of the same user, but also:
  b. sharing tabs with other TO-registered accounts.
3. Tabs tagging.
  My linear list is growing and growing every day, and often I'm reorganising my tabs based on subject into the tabs folder structure. Nice, but it's one dimensional: e.g. the time perspective is lost and we can't add other tags as a way to organise the tabs either. 

Multidimensional tabs tagging of 'Tab Windows', 'Tab Groups', and maybe even individual tabs will allow for sorting tabs along different axes: if it would be allowed to tag 'tabs' and then sort on tag type, I believe searching tabs in a vastly growing 'tab archive' will become more like a knowledge management tool. 

------------------------------------------------------------

So, I forgot about the tabs grouping idea. Would be great if the grouping in link-map (using the "Window" node reflecting one window or tab group) is reflected in the chrome tab-group. That would give the usability another tremendous boost!

**Reply by GarinZ** [reply]
Thank you very much for providing me with such rich suggestions. I also welcome you to continue to provide any form of feedback in your preferred way.

After carefully reading your feedback I now have a clearer understanding of your ideas for the tag grouping function. According to my current design, each node has its own **properties** (such as: type/createTime/lastActiveTime/url/isClosed...), and these **properties** can be used to apply sorting/filtering/grouping functions according to certain **rules**. I think our idea is close.

About Tab Group support in Chrome, this feature is also what I’m working on. Link Map will enable to bidirectional bind with the tab group in browser and tree. It won’t wait too long.
